src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java

```diff
@@ -192,21 +194,21 @@ private void computeTelemetrySettings() {
     }
 
     private void reportTelemetry() {
-        TelemetryMessageBuilder.ActionMessage telemetry = TelemetryService.instance().getBuilder().action(TelemetryService.NAME_PREFIX_MISC + "login");
+        TelemetryMessageBuilder.ActionMessage telemetry = instance().getBuilder().action(NAME_PREFIX_MISC + "login");
         try {
             ClusterInfo info = ClusterHelper.getClusterInfo(client);
             telemetry.property(KUBERNETES_VERSION, info.getKubernetesVersion());
             telemetry.property(IS_OPENSHIFT, Boolean.toString(info.isOpenshift()));
             telemetry.property(OPENSHIFT_VERSION, info.getOpenshiftVersion());
-            telemetry.send();
+            asyncSend(telemetry);
         } catch (RuntimeException e) {
             // do not send telemetry when there is no context ( ie default kube URL as master URL )
             try {
                 //workaround to not send null values
                 if (e.getMessage() != null) {
```

Comment:
after looking at the latest telemetry code, this if/then can be totally removed and be replaced by just  asyncSend(telemetry.error(e)); and maybe remove also the redundant try/catch